### PR TITLE
feat(rollapp): allow supplying bech32 and genesis checksum in update rather than create

### PR DESCRIPTION
**Pull request by danwt**
## Description

----

Closes https://github.com/dymensionxyz/dymension/issues/1088

**All** items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow-up issues.

PR review checkboxes:

I have...

- [ ]  Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ]  Targeted PR against the correct branch
- [ ]  included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ]  Linked to the GitHub issue with discussion and accepted design
- [ ]  Targets only one GitHub issue
- [ ]  Wrote unit and integration tests
- [ ]  Wrote relevant migration scripts if necessary
- [ ]  All CI checks have passed
- [ ]  Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code)
- [ ]  Updated the scripts for local run, e.g genesis_config_commands.sh if the PR changes parameters
- [ ]  Add an issue in the [e2e-tests repo](https://github.com/dymensionxyz/e2e-tests) if necessary

SDK Checklist
- [ ] Import/Export Genesis
- [ ] Registered Invariants
- [ ] Registered Events
- [ ] Updated openapi.yaml
- [ ] No usage of go `map`
- [ ] No usage of `time.Now()`
- [ ] Used fixed point arithmetic and not float arithmetic
- [ ] Avoid panicking in Begin/End block as much as possible
- [ ] No unexpected math Overflow
- [ ] Used `sendCoin` and not `SendCoins`
- [ ] Out-of-block compute is bounded
- [ ] No serialized ID at the end of store keys
- [ ] UInt to byte conversion should use BigEndian

Full security checklist [here](https://www.faulttolerant.xyz/2024-01-16-cosmos-security-1/)

----;

For Reviewer:

- [ ]  Confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ]  Reviewers assigned
- [ ]  Confirmed all author checklist items have been addressed

---;

After reviewer approval:

- [ ]  In case the PR targets the main branch, PR should not be squash merge in order to keep meaningful git history.
- [ ]  In case the PR targets a release branch, PR must be rebased.

## Summary by CodeRabbit

- **New Features**
	- Added a Bech32 prefix field to enhance rollapp information in transactions.
	- Introduced a flag for Bech32 prefix during rollapp creation and updates.
	- Updated commands for creating and updating rollapps to include the new Bech32 prefix parameter.

- **Bug Fixes**
	- Enhanced error handling in rollapp validation processes.

- **Tests**
	- Improved test coverage for rollapp creation and updates, particularly regarding the new Bech32 prefix functionality.

### CODE REVIEW

File: x/rollapp/keeper/rollapp.go [hunk 1]
```diff
@@ -205,8 +209,3 @@ func (k Keeper) IsRollappStarted(ctx sdk.Context, rollappId string) bool {
 	_, found := k.GetLatestStateInfoIndex(ctx, rollappId)
 	return found
 }
-
-func (k Keeper) IsRollappSealed(ctx sdk.Context, rollappId string) bool {
-	rollapp, found := k.GetRollapp(ctx, rollappId)
-	return found && rollapp.Sealed
```
Comment (danwt) [inline x/rollapp/keeper/rollapp.go:211]: it was unused
is that a bug @zale144 ?

Comment (danwt) [reply x/rollapp/keeper/rollapp.go:211]: https://github.com/dymensionxyz/dymension/issues/1090

File: x/rollapp/client/cli/tx_update_rollapp.go [hunk 2]
```diff
@@ -12,7 +12,7 @@ import (
 
 func CmdUpdateRollapp() *cobra.Command {
 	cmd := &cobra.Command{
-		Use:     "update-rollapp [rollapp-id] [init-sequencer] [genesis_checksum] [metadata]",
+		Use:     "update-rollapp [rollapp-id] [init-sequencer] [genesis_checksum] [metadata] [bech32-prefix]",
```
Comment (omritoptix) [inline x/rollapp/client/cli/tx_update_rollapp.go:15]: I'd position bech32 before the metadata as it's required

File: x/rollapp/types/rollapp.go [hunk 3]
```diff
@@ -68,8 +69,10 @@ func (r Rollapp) ValidateBasic() error {
 		return errorsmod.Wrap(ErrInvalidInitialSequencer, err.Error())
 	}
 
-	if err = validateBech32Prefix(r.Bech32Prefix); err != nil {
-		return errorsmod.Wrap(ErrInvalidBech32Prefix, err.Error())
+	if r.Sealed || r.Bech32Prefix != "" {
```
Comment (omritoptix) [inline x/rollapp/types/rollapp.go:72]: not sure I understand why we check for if sealed here? 

this validate basic is called when:
1. rollapp creation (don't need to validate sealed cause for sure it's not on creation)
2. rollapp update (if we try to update bech32, it should fail before in the `UpdatingImmutableValues() && current.Sealed`

File: x/rollapp/client/cli/tx_update_rollapp.go (hunk 2 above)
Comment (danwt) [reply x/rollapp/client/cli/tx_update_rollapp.go:15]: done

File: x/rollapp/types/rollapp.go (hunk 3 above)
Comment (danwt) [reply x/rollapp/types/rollapp.go:72]: I guess I was being too strict (for import genesis)

I'll remove

**Review by omritoptix — APPROVED** [review]

**Pull request merged**